add import and export in index

TheRealAstoo opened the pull request:
## Description

I'm adding the import and export of the function `generateJWE` in the index.

## Motivation and Context

I could not use it in Sparta-API, so I came here 😄 

## How Has This Been Tested?

I did not test it.

## Types of changes

<!--- What types of changes does your code introduce? Stroke -->
- [x] Chore (non-breaking change which refactors / improves the existing code base)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  change)

## Checklist:

<!--- If you're unsure about any of these, don't hesitate to ask. We're -->
<!--- here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] My change requires a change to a package version.
- [x] I have updated the `package.json` version accordingly.
- [x] I have updated the `CHANGELOG.md` version accordingly.
- [x] I have read the [**CONTRIBUTING**][CONTRIBUTING_FILE] document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

[CONTRIBUTING_FILE]: https://github.com/fewlinesco/node-web-libraries/blob/master/CONTRIBUTING.md

Fenntasy commented:
Task linked: [CU-jb6590 [chore] - add generateJWE export](https://app.clickup.com/t/jb6590)

TheRealAstoo commented:
Big changes here 😎

Fuzznimp commented:
You need to bump the version and add the entry to the changelog.

Oluane commented:
You need change the version in `package.json`, add a line in the `CHANGELOG` about your changes and then publish a new version of the package 🙂

TheRealAstoo commented:
Thanks for your answers guys 😃

TheRealAstoo commented:
Is it ok for the changelog ? It was my first 😄

TheRealAstoo commented:
Do I need to do something else to push it on NPM or it's automatic / done by one of you only?

Fuzznimp commented:
> Is it ok for the changelog ? It was my first 😄

No worries, we'll comment if needed.

> Do I need to do something else to push it on NPM or it's automatic / done by one of you only?

You need to:
- checkout to main
- pull main
- `npm longin` (credentials are in the team vault)
- `npm publish --access public` (from the package root folder)

Fuzznimp commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,55 @@
+
+<!--- Describe your changes in detail -->
+
+## Related Issue
+
+<!--- This project only accepts pull requests related to open issues -->
+<!--- If suggesting a new feature or change, please discuss it in an -->
+<!--- issue first -->
+<!--- If fixing a bug, there should be an issue describing it with -->
+<!--- steps to reproduce -->
+<!--- Please link to the issue here with the prefix or resolve -->
+<!--- keyword -->
```
This is not needed, as the bot link it through the task ID from the branch's name.

TheRealAstoo replied:
Can be cool for reviewers to have a link?

Fuzznimp replied:
The link will be added automatically by the bot
![image](https://user-images.githubusercontent.com/31956107/117166229-7865cf00-adc6-11eb-884a-cbda482de4cb.png)

TheRealAstoo replied:
True. I change it 👍

Fuzznimp commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,55 @@
+<!--- you ran to see how your change affects other areas of the -->
+<!--- code, etc. -->
+
+## Screenshots
+
+<!--- if appropriate, otherwise the section can be removed -->
+
```
This is not needed as there is no frontend.

Fuzznimp commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,55 @@
+- Breaking change (fix or feature that would cause existing functionality to
+  change)
+
+## Checklist:
+
+<!--- Go over all the following points, and replace the `:red_circle:` in all -->
+<!--- the lines with a :white_check_mark: when relevant. -->
+<!--- If you're unsure about any of these, don't hesitate to ask. We're -->
+<!--- here to help! -->
+- :red_circle: My code follows the code style of this project.
+- :red_circle: My change requires a change to the documentation.
+- :red_circle: I have updated the documentation accordingly.
+- :red_circle: I have read the [**CONTRIBUTING**][CONTRIBUTING_FILE] document.
+- :red_circle: I have added tests to cover my changes.
+- :red_circle: All new and existing tests passed.
```
You should use this checklist https://github.com/fewlinesco/node-web-libraries/blob/main/.github/PULL_REQUEST_TEMPLATE.md/

TheRealAstoo replied:
I prefer it to. Faster to look at or change something after.

Meyclem replied:
![image](https://user-images.githubusercontent.com/29123047/117166629-d72b4880-adc6-11eb-9c95-755919bf4e65.png)
😏

TheRealAstoo replied:
Hum, my bad. I took the Sparta version (that we should maybe change then ?)

Fuzznimp replied:
`sparta-monorepo` and `connect-js` are not the same kind, so I'd say no. I mean, you'd still have to provide screenshots, but we don't do versionning on `sparta-monorepo`.

TheRealAstoo replied:
Oh, I was talking about the circles vs the checks 👍

Fuzznimp commented on client/CHANGELOG.md:
```diff
@@ -2,6 +2,10 @@
 All notable changes to this project will be documented in this file.
 
+## [0.4.5] - 2021-05-05
+
+- Added the export of `generateJWE`
+
```
```suggestion

- Added the export of `generateJWE`.

```

Fuzznimp commented on client/CHANGELOG.md:
```diff
@@ -2,6 +2,10 @@
 
 All notable changes to this project will be documented in this file.
 
+## [0.4.5] - 2021-05-05
+
```
```suggestion

## [0.4.6] - 2021-05-05

```

Meyclem commented on client/CHANGELOG.md:
```diff
@@ -2,9 +2,9 @@
+## [0.4.6] - 2021-05-05
 
-- Added the export of `generateJWE`
+- Added the export of `generateJWE`.
```
```suggestion
- Added the export of `generateJWE`. This was missing from `0.4.4`
```
Otherwise it could raise questions. This way we give all the need information about why this change is needed.

Meyclem commented:
And you can also update the PR description 🙂

TheRealAstoo commented:
> description

With the new template?

Meyclem commented on client/CHANGELOG.md:
```diff
@@ -2,6 +2,10 @@
 
+## [0.4.6] - 2021-05-05
+
+- Added the export of `generateJWE`. This was missing from `0.4.4`
```
```suggestion
- Added the export of `generateJWE`. This was missing in the `0.4.4` version.
```
You were too quick to make the change ;)

TheRealAstoo replied:
😄

Meyclem commented:
> > And you can also update the PR description 🙂
> 
> With the new template?

Yes

TheRealAstoo commented:
> > > And you can also update the PR description 🙂
> > 
> > 
> > With the new template?
> 
> Yes

Just seen something weird in the template do we put circles or cross? 🤔 : 

<!--- Go over all the following points, and replace the `:red_circle:` in all -->
<!--- the lines with a :white_check_mark: when relevant. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're -->
<!--- here to help! -->
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] My change requires a change to a package version.
- [ ] I have updated the `package.json` version accordingly.
- [ ] I have updated the `CHANGELOG.md` version accordingly.
- [ ] I have read the [**CONTRIBUTING**][CONTRIBUTING_FILE] document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

Fuzznimp commented:
> Just seen something weird in the template do we put circles or x check? 🤔 

Just put x.

Fuzznimp commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,41 @@
+- [ ] I have read the [**CONTRIBUTING**][CONTRIBUTING_FILE] document.
+- [ ] I have added tests to cover my changes.
+- [ ] All new and existing tests passed.
+
+[CONTRIBUTING_FILE]: https://github.com/fewlinesco/node-web-libraries/blob/master/CONTRIBUTING.md
```
This points to an unrelated repo. I'd say we remove it for now, until we have a proper file for the monorepo.
